pkg/monitor/monitor.go

```diff
@@ -132,39 +172,103 @@ func (monitor *controllerMonitor) createControllerCRD() (*v1beta1.AntreaControll
 			},
 		},
 	}
-	klog.V(2).Infof("Creating controller monitor CRD %v", controllerCRD)
+	klog.V(2).Infof("Creating controller monitoring CRD %v", controllerCRD)
 	return monitor.client.ClusterinformationV1beta1().AntreaControllerInfos().Create(controllerCRD)
 }
 
-// TODO: Update network policy related fields when the upstreaming is ready
+// updateControllerCRD updates all the fields of existing monitoring CRD
 func (monitor *controllerMonitor) updateControllerCRD(controllerCRD *v1beta1.AntreaControllerInfo) (*v1beta1.AntreaControllerInfo, error) {
-	klog.V(2).Infof("Updating controller monitor CRD %v", controllerCRD)
+	controllerCRD.Version = version.GetFullVersion()
+	controllerCRD.PodRef = monitor.GetSelfPod()
+	controllerCRD.NodeRef = monitor.GetSelfNode()
+	controllerCRD.ServiceRef = monitor.GetService()
+	controllerCRD.ControllerConditions = []v1beta1.ControllerCondition{
+		{
+			Type:              v1beta1.ControllerHealthy,
+			Status:            v1.ConditionTrue,
+			LastHeartbeatTime: metav1.Now(),
+		},
+	}
+	klog.V(2).Infof("Updating controller monitoring CRD %v", controllerCRD)
+	return monitor.client.ClusterinformationV1beta1().AntreaControllerInfos().Update(controllerCRD)
+}
+
+// partialUpdateControllerCRD only updates the variables.
+func (monitor *controllerMonitor) partialUpdateControllerCRD(controllerCRD *v1beta1.AntreaControllerInfo) (*v1beta1.AntreaControllerInfo, error) {
 	controllerCRD.ControllerConditions = []v1beta1.ControllerCondition{
 		{
 			Type:              v1beta1.ControllerHealthy,
 			Status:            v1.ConditionTrue,
 			LastHeartbeatTime: metav1.Now(),
 		},
 	}
+	klog.V(2).Infof("Partially updating controller monitoring CRD %v", controllerCRD)
 	return monitor.client.ClusterinformationV1beta1().AntreaControllerInfos().Update(controllerCRD)
 }
 
+func (monitor *controllerMonitor) deleteStaleAgentCRDs() {
+	nodes, err := monitor.nodeInformer.Lister().List(labels.Everything())
+	if err != nil {
+		klog.Error("Failed to list all the nodes")
+		return
+	}
+	crds := monitor.listAgentCRD()
+	if crds == nil {
+		klog.Info("Failed to list agent monitoring CRDs")
```

Comment:
This is duplicate with L238. I think it should be at least warning? even error?
Go's convention would let `listAgentCRD` return `(*v1beta1.AntreaAgentInfoList, error)` and you can then log the error here. The function name should be `listAgentCRDs`

Reply:
OK, got it.